### ISSUE TITLE
Feature request: on iOS, open downloaded file in new tab

### ISSUE THREAD

**Issue by mjbaldwin** [RobinLinus/snapdrop#195]
First of all, this app is _amazing_ -- thank you. I use AirDrop a lot to transfer PDF's between my Mac and my iPad, and for whatever reason it fails about half the time until I restart both devices which is a pain. Snapdrop saves the day!

However, there's a bit of an inconvenience which is that whenever I send a PDF from my Mac to my iPad, after it transfers it shows a popup with the option "download", and clicking "download" opens the PDF in the current tab that Snapdrop was it. I then share it to my PDF reader... but then I need to press the "back" button on my iPad to go back to Snapdrop (to transfer another file), and often it's no longer connected (no longer shows up on my Mac), and I need to reload it.

Now it's definitely not the end of the world. :) But it would definitely be a smoother workflow if there were any way to 1) force the PDF to be sent directly to a "share" dialog rather than opening in the browser, or else 2) if that can't be done, then see if there's a way to force the PDF to be opened in a new tab (or even as a tab opened in the background)? That would just make it a little bit easier to transfer several files one after another.

If iOS Safari has limitations that don't make it possible then no worries -- just thought I'd share my usage experience!

Thanks again

**Comment by Bellisario** [comment]
https://github.com/RobinLinus/snapdrop/blob/9db8c4ce8cef15ea41319846483ee526874306b3/client/scripts/ui.js#L259-L265

There is a _blank target for iOS and MacOS, but, evidently, it doesn't work corretly...

**Comment by RobinLinus** [comment]
That's true. Downloading / handling files is a bit cumbersome on iOS. Let me see if it's possible to build a workaround using the [Web Share API](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share) .

**Comment by mjbaldwin** [comment]
I did a little bit of research, and it seems like this is a known issue
with PDF's in iOS Safari, that they display rather than download:

https://bugs.webkit.org/show_bug.cgi?id=167341

It presumably applies to other content types the browser can also display
(like JPG, etc.)

There's a snippet of code on the page that shows a workaround, but I'm not
sure if it's reliable. Autodetecting file formats that Safari forces to
display rather than download and then sending to a Share dialog seems like
it would be a clever workaround if it works!
